zipline/src/hostTest/kotlin/app/cash/zipline/internal/bridge/SerialNameTest.kt

```diff
@@ -79,4 +80,26 @@ class SerialNameTest {
       serialName,
     )
   }
+
+  @Test
+  fun recursiveSerializer() {
+    val serialName = serialName(
+      "SomeType",
+      serializers = listOf(
+        serializer<SomeRecursiveType>(),
+      ),
+    )
+    assertEquals(
+      expected = "SomeType<" +
+        "app.cash.zipline.internal.bridge.SomeRecursiveType<" +
+        "app.cash.zipline.internal.bridge.SomeRecursiveType?<app.cash.zipline.internal.bridge.SomeRecursiveType?>" +
```

Comment:
I'm wondering if it's necessary to include the second copy here. If we break before recursing it won't be included, and isn't it entirely redundant?

Reply:
@JakeWharton  you are correct, it does exit, in above: 
 1. 1st serialName of **Non-Nullable** `SomeRecursiveType`.
 2. 1st serialName of **Nullable** `SomeRecursiveType`.
 3. 2nd **Recurse** serialName of **Nullable** `SomeRecursiveType`, hence exits with plain serialName of descriptor.

Reply:
Right my question was if the second copy is actually necessary or redundant

Reply:
AhUnderstood what you mean,
Yes we could just return empty string once we hit recursion instead of repeating serialName, shouldn't matter.